src/config/config.rs

```diff
@@ -45,13 +45,12 @@ pub struct Config {
     ///
     /// A map of all the types in the schema.
     #[serde(default)]
-    #[setters(skip)]
-    pub types: BTreeMap<String, Type>,
+    pub types: HashMap<String, Type>,
 
     ///
     /// A map of all the union types in the schema.
     #[serde(default, skip_serializing_if = "is_default")]
-    pub unions: BTreeMap<String, Union>,
+    pub unions: HashMap<String, Union>,
```

Comment:
HashMap doesn't guarantee order. That makes it hard to assert on tests.

Reply:
BTreeMap is slower, has average time of O(log n) while HashMap is O(1). Should we "give up" this performance boost and revert to BTreeMap?

Note that on parts that depend on ordering the HashMap is being collected into a BTreeMap, so we don't have issues with the tests in the current implementation. But I can for sure rollback this change if you prefer

Reply:
A simple function can help comparing two hashmaps

```rs

map1.len() == map2.len() && map1.keys().all(|k| map2.contains_key(k))

```

If test cases are the only place to check for equality, it won't be a big of an issue probably.